tests/conftest.py

```diff
@@ -0,0 +1,370 @@
+#   2. Add `lazy_fixture("model_key_with_kwargs")` to the param list of the `models_with_kwargs`
+#      fixture. (Where `"model_key_with_kwargs"` is the name of the fixture you created in step 1.)
+
+
+@pytest.fixture(scope="session")
+def recipe_run_with_kwargs():
+    kws = [
+        ModelKwargs(
+            request=dict(
+                recipe_id="test-recipe-0",
+                run_date="2021-01-01T00:00:00Z",
+                bakery_id=0,
+                feedstock_id=0,
+                commit="012345abcdefg",
+                version="1.0",
+                status="complete",
+                path="/path-to-dataset.zarr",
+            ),
+            blank_opts=["message"],
+        ),
+        ModelKwargs(
+            request=dict(
+                recipe_id="test-recipe-1",
+                run_date="2021-02-02T00:00:00Z",
+                bakery_id=1,
+                feedstock_id=1,
+                commit="012345abcdefg",
+                version="2.0",
+                status="complete",
+                path="/path-to-dataset.zarr",
+                message="hello",
+            ),
+        ),
+        ModelKwargs(
+            request=dict(
+                recipe_id="test-recipe-2",
+                run_date="2021-03-03T00:00:00Z",
+                bakery_id=2,
+                feedstock_id=2,
+                commit="012345abcdefg",
+                version="3.0",
+                status="complete",
+                path="/path-to-dataset.zarr",
+            ),
+            blank_opts=["message"],
+        ),
+    ]
+    return ModelFixture(MODELS["recipe_run"], kws)
+
+
+@pytest.fixture(
+    scope="session", params=[lazy_fixture("recipe_run_with_kwargs")],
+)
+def models_with_kwargs(request):
+    return request.param
+
+
+# CRUD function fixtures ------------------------------------------------------------------
+
+# Create ----------------------------------------------------------------------------------
+
+
+class CreateFixtures:
+    """Fixtures for ``TestCreate``"""
+
+    @pytest.fixture
+    def model_to_create(self, models_with_kwargs):
+        kw_0, _, _ = models_with_kwargs.kwargs
+        return models_with_kwargs.models, kw_0.request, kw_0.blank_opts
+
+    @pytest.fixture
+    def create_with_db(self) -> Callable:
+        return DatabaseCRUD.create_with_db
+
+    @pytest.fixture
+    def create_with_abstraction(self) -> Callable:
+        return AbstractionCRUD.create_with_abstraction
+
+    @pytest.fixture
+    def create_with_client(self) -> Callable:
+        return ClientCRUD.create_with_client
+
+    @pytest.fixture
+    def create_with_cli(self) -> Callable:
+        return CommandLineCRUD.create_with_cli
+
+    @pytest.fixture(
+        params=[
+            lazy_fixture("create_with_db"),
+            lazy_fixture("create_with_abstraction"),
+            lazy_fixture("create_with_client"),
+            lazy_fixture("create_with_cli"),
+        ],
+    )
+    def create_func(self, request):
```

Comment:
Ok nevermind, I found it. This is pretty complicated. 🤯

Reply:
😄 Yes, it is! Which I see more clearly upon returning to it after a break.

Simplified in https://github.com/pangeo-forge/pangeo-forge-orchestrator/pull/6/commits/72a24caa6252c387d1f9e2597be224ddaec07cac